node/hack/repair_terra/repair.go

```diff
@@ -0,0 +1,428 @@
+	var address vaa.Address
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return address, err
+	}
+	copy(address[:], res)
+	return address, nil
+}
+
+// StringToHash convert string into transaction hash
+func StringToHash(value string) (eth_common.Hash, error) {
+	var hash eth_common.Hash
+	res, err := hex.DecodeString(value)
+	if err != nil {
+		return hash, err
+	}
+	copy(hash[:], res)
+	return hash, nil
+}
+
+func main() {
+	flag.Parse()
+	fmt.Println("terraAddr", *terraAddr)
+
+	ctx := context.Background()
+
+	currentHeight, err := getCurrentHeight()
+	if err != nil {
+		log.Fatalf("Failed to get current height: %v", err)
+	}
+	log.Printf("Current height: %d", currentHeight)
+
+	missingMessages := make(map[uint64]bool)
+
+	conn, err, admin := getAdminClient(ctx, *adminRPC)
+	defer conn.Close()
+	if err != nil {
+		log.Fatalf("failed to get admin client: %v", err)
+	}
+
+	var TerraEmitter = struct {
+		ChainID vaa.ChainID
+		Emitter string
+	}{vaa.ChainIDTerra, "0000000000000000000000007cf7b764e38a0a5e967972c1df77d432510564e2"}
+
+	log.Printf("Requesting missing messages for %s)", TerraEmitter.Emitter)
+
+	msg := nodev1.FindMissingMessagesRequest{
+		EmitterChain:   uint32(vaa.ChainIDTerra),
+		EmitterAddress: TerraEmitter.Emitter,
+		RpcBackfill:    true,
+		BackfillNodes:  common.PublicRPCEndpoints,
+	}
+	resp, err := admin.FindMissingMessages(ctx, &msg)
+	if err != nil {
+		log.Fatalf("failed to run find FindMissingMessages RPC: %v", err)
+	}
+
+	msgs := make([]*db.VAAID, len(resp.MissingMessages))
+	for i, id := range resp.MissingMessages {
+		fmt.Println(id)
+		vId, err := db.VaaIDFromString(id)
+		if err != nil {
+			log.Fatalf("failed to parse VAAID: %v", err)
+		}
+		msgs[i] = vId
+	}
+
+	if len(msgs) == 0 {
+		log.Printf("No missing messages found for %s", TerraEmitter)
+		return
+	}
+
+	lowest := msgs[0].Sequence
+	highest := msgs[len(msgs)-1].Sequence
+
+	log.Printf("Found %d missing messages for %s: %d - %d", len(msgs), TerraEmitter, lowest, highest)
+
+	for _, msg := range msgs {
+		missingMessages[msg.Sequence] = true
+	}
+
+	// missingMessages[28467] = true
+	// lowest := 28467
+
+	log.Printf("Starting search for missing sequence numbers...")
+	offset := 0
+	var firstTime bool = true
+	for ((offset > 0) || firstTime) {
+		firstTime = false
+		client := &http.Client{ 
+			Timeout: time.Second * 5, 
+		}
+		resp, err := client.Get(fmt.Sprintf("https://fcd.terra.dev/v1/txs?offset=%d&limit=100&account=%s", offset, *terraAddr))
+		if err != nil {
+			fmt.Errorf("failed to get log: %w", err)
+			continue
+		}
+		blocksBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Errorf("failed to read log: %w", err)
+			continue
+		}
+		resp.Body.Close()
+
+		blockJSON := string(blocksBody)
+		// fmt.Println("blockJSON", blockJSON)
+		if !gjson.Valid(blockJSON) {
+			fmt.Errorf("invalid JSON response")
+			continue
+		}
+		next := gjson.Get(blockJSON, "next")
+		fmt.Println("next block", next.Int())
```

Comment:
If this is not debugging related, make it a log.xxx please

Reply:
Changed